doc/http-api-guide.md

```diff
@@ -250,3 +270,94 @@ EOF
   "status":0,
   "message":"OK" }
 ```
+
+
+## Subscription data
+
+Clients may access subscription data via native WebSockets, SocketIO, or
+long-polling over HTTP.
+
+The preferred method to access subscription data is over native WebSockets.
+WebSockets is a web standard for receiving event-driven messages without having
+to resort to long-polling.  SocketIO is a higher-level library built on top of
+WebSockets, which has transparent long-polling fallback support, automatic
+reconnection, and the ability to define custom events; however, it is not a
+standard and language support is limited to JavaScript at the time of this
+writing.  For compatability reasons, long-polling is also supported for clients
+that are unable to utilize the WebSocket protocol.
+
+The following sections discuss the details for subscription data over native
+WebSockets, SocketIO, and long-polling over HTTP.
+
+### WebSocket and SocketIO message types
+
+When accesing subscription data via native WebSockets or SocketIO, each
+mechanism shares the same message types.  Note that when using WebSockets, that
+the message comes in the form:
+
+```javascript
+{
+    'type': /* message type */,
+    'data': /* message data */
+}
+```
+
+### Client to Server
```

Comment:
We should add some notes about which is *type* and which is *data* for the following section.